### PR TITLE
Add user's org to the token

### PR DESCRIPTION
**Pull request by diversemix**
# Description

- add org to token
- update tests
- fix is_admin to always be True as per real life

## Proposed version

Please select the option below that is most relevant from the list below. This
will be used to generate the next tag version name during auto-tagging.

- [ ] Skip auto-tagging
- [x] Patch
- [ ] Minor version
- [ ] Major version

Visit the [Semver website](https://semver.org/#summary) to understand the
difference between `MAJOR`, `MINOR`, and `PATCH` versions.

Notes:

- If none of these options are selected, auto-tagging will fail
- Where multiple options are selected, the most senior option ticked will be
  used -- e.g. Major > Minor > Patch
- If you are selecting the version in the list above using the textbox, make
  sure your selected option is marked `[x]` with no spaces in between the
  brackets and the `x`

## Type of change

Please select the option(s) below that are most relevant:

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] GitHub workflow update
- [ ] Documentation update
- [ ] Remove legacy code
- [ ] Dependency update

## How Has This Been Tested?

Please describe the tests that you added to verify your changes.

## Reviewer Checklist

- [ ] The PR represents a single feature (small drive-by fixes are also ok)
- [ ] The PR includes tests that are sufficient for the level of risk
- [ ] The code is sufficiently commented, particularly in hard-to-understand areas
- [ ] Any required documentation updates have been made
- [ ] Any TODOs added are captured in future tickets
- [ ] No FIXMEs remain

### CODE REVIEW

File: app/service/authentication.py [hunk 1]
```diff
@@ -59,6 +60,11 @@ def authenticate_user(email: str, password: str) -> str:
             _LOGGER.error(f"Failed login attempt as inactive {email}")
             raise AuthenticationError(f"User {email} is marked as not active.")
 
+        org_id = app_user_repo.get_org_id(db, cast(str, user.email))
+        if org_id is None:
+            _LOGGER.error(f"Failed login attempt, org not found {email}")
```
Comment (THOR300) [inline app/service/authentication.py:65]: `...not found for {email}` might read better

File: tests/integration_tests/conftest.py [hunk 2]
```diff
@@ -218,34 +220,34 @@ def rollback_event_repo(monkeypatch, mocker):
...
 def superuser_header_token() -> Dict[str, str]:
-    a_token = token_service.encode("super@cpr.org", True, {})
+    a_token = token_service.encode("super@cpr.org", ORG_ID, True, {})
     headers = {"Authorization": f"Bearer {a_token}"}
     return headers
 
 
 @pytest.fixture
 def user_header_token() -> Dict[str, str]:
-    a_token = token_service.encode("cclw@cpr.org", False, {"is_admin": False})
```
Comment (THOR300) [inline tests/integration_tests/conftest.py:228]: I'm interested why this `org_id` update now means all users are admins?

**Review by THOR300 — DISMISSED** [review]
Non blocking query about why all users are admins now! 

And great that the org id is just an integer and thus non-sensitive data! 

Is it a concern that we are encoding email addresses?

**Comment by katybaulch** [comment]
> Non blocking query about why all users are admins now!
> 
> And great that the org id is just an integer and thus non-sensitive data!
> 
> Is it a concern that we are encoding email addresses?

Hey, it's two separate changes! So we want to add the org_id to the user token so we can access it easily in the backend. The driveby fix here is to make bump the test users to admin to reflect the real life database. Admin users are able to use the admin service but are restricted to CRUDding entities belonging to the same organisation as themselves, whereas superuser can CRUD all entities across all organisations.

Not at all luckily, as it's all in the JWT token which can't be decoded without the secret key!

**Comment by THOR300** [comment]
> > Non blocking query about why all users are admins now!
> > And great that the org id is just an integer and thus non-sensitive data!
> > Is it a concern that we are encoding email addresses?
> 
> Hey, it's two separate changes! So we want to add the org_id to the user token so we can access it easily in the backend. The driveby fix here is to make bump the test users to admin to reflect the real life database. Admin users are able to use the admin service but are restricted to CRUDding entities belonging to the same organisation as themselves, whereas superuser can CRUD all entities across all organisations.
> 
> Not at all luckily, as it's all in the JWT token which can't be decoded without the secret key!

I only ask as I was doing some quick reading around as I've not personally used JWT's in this much detail before, always been a consumer of them rather than a creator! 

"When it comes to access tokens, they can easily be replaced by opaque tokens (as mentioned in the previous section), but the ID token is always a JWT. This means that you should put extra care into what is available in the JWT so that no sensitive data is unintentionally revealed. It will be much safer for your UI client to call the user info endpoint and get the user's data from there instead of keeping it directly in the ID token." 

My understanding of this was that if we want the user data we can just get that after the frontend receives the token? 

https://curity.io/resources/learn/jwt-best-practices/

**Comment by diversemix** [comment]
@THOR300 ... you are of course right and this statement (which I said to Katy in the middle of coding) is wrong: "Not at all luckily, as it's all in the JWT token which can't be decoded without the secret key!" ... what I should have said is that its impossible for the user to fake a token - i.e. the user or anyone who has the token can decode it ... but they cannot fake a token without having the private key... so that for instance they cannot set themselves to belong to another organisation. 

Second point - we could use the user's db id to identify them rather than their email - but we don't use "real" emails  anyway. This is a good point if we should do - so gonna raise a ticket - thanks.

**Comment by diversemix** [comment]
See https://linear.app/climate-policy-radar/issue/PDCT-1174/when-using-real-emails-do-not-use-in-the-jwt

**Review by THOR300 — APPROVED** [review]
LGTM

**Pull request merged**